UefiCpuPkg/CpuDxe/CpuMp.c

```diff
@@ -661,7 +661,12 @@ InitializeMpExceptionStackSwitchHandlers (
   UINT8                           *Buffer;
 
   SwitchStackData = AllocateZeroPool (mNumberOfProcessors * sizeof (EXCEPTION_STACK_SWITCH_CONTEXT));
-  ASSERT (SwitchStackData != NULL);
+  if (SwitchStackData == NULL) {
+    DEBUG ((DEBUG_ERROR, "%a Failed to allocate buffer for SwitchStackData\n", __func__));
+    ASSERT (FALSE);

```

Comment:
I suggest keeping the original assert since it will produce a more relevant message.

Current

```text
PlatformSmbiosDriverEntryPoint Failed to allocate buffer for SwitchStackData

DXE_ASSERT!: [CpuDxe] c:\src\edk2\UefiCpuPkg\CpuDxe\CpuMp.c (666): ((BOOLEAN)(0==1))
```

Proposed

```text
PlatformSmbiosDriverEntryPoint Failed to allocate buffer for SwitchStackData

DXE_ASSERT!: [CpuDxe] c:\src\edk2\UefiCpuPkg\CpuDxe\CpuMp.c (666): SwitchStackData != ((void *) 0)
```

```suggestion
    ASSERT (SwitchStackData != NULL);
```

Reply:
Updated.